### PR TITLE
Add pyramid page and modularize pyramid assets

### PR DESCRIPTION
**Pull request by gustavoalfaro**
Add pyramid 3d in react and modularize assets for rendering

## Screenshots

image

## Video
http://recordit.co/LfaqoT9DYT

### CODE REVIEW

**Review by mckelvey — APPROVED** [review]
This looks good @gustavoalfaro — I just had some questions. :)

File: pages/pyramid/index.js [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+import { linesFs, linesVs } from './shapes/lines'
+import { innerShapeFs, innerShapeVs } from './shapes/innerShape'
+import {
+  outerShapeBackFaceFs,
+  outerShapeBackFaceVs,
+} from './shapes/outerShapeBackFace'
+
+const PyramidPage = () => {
+  useEffect(() => {
+    function clamp(val, lo, hi) {
```
[the remark this answers is not in the document]
Comment (gustavoalfaro) [reply pages/pyramid/index.js:15]: The state of the code atm

File: pages/pyramid/index.js [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+      endValue: 0,
+    }
+    const spinAnimation = {
+      elapsedMilliseconds: 0,
+      durationMilliseconds: 800 * 2,
+      beginValue: 0,
+      endValue: 0,
+    }
+
+    const canvas = document.querySelector('#pyramidCanvas')
```
[the remark this answers is not in the document]
Comment (gustavoalfaro) [reply pages/pyramid/index.js:317]: yup, I was kinda submitting this as draft, so I'll take care of that now, thanks for the comments!

File: pages/pyramid/index.js [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+      const height = Math.trunc(canvas.clientHeight * pixelRatio)
+      const needResize = canvas.width !== width || canvas.height !== height
+      if (needResize) {
+        renderer.setSize(width, height, false)
+      }
+      return needResize
+    }
+
+    const whiteLightPosition = new THREE.Vector3(20, 40, 3)
+    const whiteLightAmount = 0.25
```
[the remark this answers is not in the document]
Comment (mckelvey) [reply pages/pyramid/index.js]: Awesome!

File: pages/pyramid/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mckelvey) [reply pages/pyramid/index.js]: 👍

File: pages/pyramid/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mckelvey) [reply pages/pyramid/index.js]: Cool... expected that, but worth confirming.

**Pull request merged**